frontend/javascripts/oxalis/view/left-border-tabs/histogram_view.js

```diff
@@ -185,6 +191,79 @@ class Histogram extends React.PureComponent<HistogramProps, HistogramState> {
     500,
   );
 
+  getClippingValues = async (layerName: string, threshold: number = 0.05) => {
+    const { elementClass } = getLayerByName(Store.getState().dataset, layerName);
+    const [TypedArrayClass] = getConstructorForElementClass(elementClass);
+
+    const [cuboidXY, cuboidXZ, cuboidYZ] = await Promise.all([
+      api.data.getDataForViewport(OrthoViews.PLANE_XY, layerName),
+      api.data.getDataForViewport(OrthoViews.PLANE_XZ, layerName),
+      api.data.getDataForViewport(OrthoViews.PLANE_YZ, layerName),
+    ]);
+    const dataForAllViewports = new TypedArrayClass(
+      cuboidXY.length + cuboidXZ.length + cuboidYZ.length,
+    );
+
+    dataForAllViewports.set(cuboidXY);
+    dataForAllViewports.set(cuboidXZ, cuboidXY.length);
+    dataForAllViewports.set(cuboidYZ, cuboidXY.length + cuboidXZ.length);
+
+    const localHist = {};
```

Comment:
Using a mapping structure here (and for `accumulator`) is a good idea to avoid large arrays. However, JS objects have the annoying property that numeric keys are automatically converted to strings. This is why you needed the `Number` casting below in the code. This can lead to subtle bugs which is why I'd recommend to use `new Map()` here. The interface looks like this:

```
const localHist = new Map();
localHist.set(3, 10);
localHist.get(3) == 10;
```

Then, you can also remove the `Number(...)` conversions below.